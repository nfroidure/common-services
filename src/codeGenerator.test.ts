import initCodeGenerator from './codeGenerator';
import type { RandomService } from './random';

describe('codeGenerator', () => {
  const log = jest.fn();
  const random = jest.fn() as jest.Mock<number, RandomService[]>;

  beforeEach(() => {
    log.mockReset();
    random.mockReset();
  });

  test('should work', async () => {
    random.mockReturnValueOnce(0.5);
    const codeGenerator = await initCodeGenerator({
      random,
      log,
    });
    const code = await codeGenerator();

    expect({
      code,
      logCalls: log.mock.calls,
      randomCalls: random.mock.calls,
    }).toMatchSnapshot();
  });
});
