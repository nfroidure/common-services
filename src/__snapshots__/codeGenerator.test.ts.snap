// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codeGenerator should work 1`] = `
{
  "code": "S",
  "logCalls": [
    [
      "debug",
      "↪️ - Code generation service Initialized!",
    ],
    [
      "debug",
      "↪️ - Generated a new code:",
      "S",
    ],
  ],
  "randomCalls": [
    [],
    [],
    [],
    [],
    [],
    [],
  ],
}
`;
